frontend/app/profile/page.tsx

```diff
@@ -0,0 +1,367 @@
+                  {renderEditableField("Given Name", "firstName")}
+                  {renderEditableField("Surname", "lastName")}
+                  {renderEditableField("Email", "email", "email")}
+                  {renderEditableField("Phone Number", "phoneNumber", "tel")}
+                  {renderEditableField("Location", "location")}
+                  {renderEditableField("Date of Birth", "birthday", "date")}
+                  {renderEditableField("Password", "password", "password")}
+                </div>
+                <div className="flex justify-end mt-4">
+                  <button
+                    type="button"
+                    onClick={handleSaveClick}
+                    className="text-white bg-green-500 hover:bg-green-600 focus:ring-4 focus:ring-green-300 font-medium rounded-full text-med px-6 py-2 z-50"
+                  >
+                    Save
+                  </button>
+                  <button
+                    type="button"
+                    onClick={handleEditClick}
+                    className="ml-4 text-white bg-red-500 hover:bg-red-600 focus:ring-4 focus:ring-red-300 font-medium rounded-full text-med px-6 py-2 z-50"
+                  >
+                    Cancel
+                  </button>
+                </div>
+              </Dialog.Panel>
+            </Transition.Child>
+          </div>
+        </div>
+      </Dialog>
+    </Transition>
+  );
+
+  const editedAge = calculateAge(editedData.birthday);
+
+  return (
+    <div className="w-screen flex justify-center">
+      <div className="w-screen mx-10 sm:mx-0 sm:w-[400px] md:w-[700px] lg:w-[1000px] xl:w-[1000px] 3xl:w-[1400px]">
+        <div className="relative mt-[100px] mb-[5%] grid grid-cols-1 md:grid-cols-2 md:gap-x-[5vw] lg:gap-x-[3vw]">
+          <div className="flex justify-center md:justify-start col-start-1 col-span-1 md:col-start-2 md:row-span-1 md:row-start-1">
+            <div
+              className="flex justify-center mb-5 md:mb-6 text-3xl lg:text-4xl"
+              style={{ fontWeight: 600 }}
+            >
+              {initialProfileData.firstName}'s Profile
+            </div>
+          </div>
+          <div className="flex flex-col">
+            <div className="flex justify-center w-full h-fit col-start-1 col-span-1 md:row-start-1">
+              <div
+                className="lg:flex-row border border-gray-500 lg:h-60 w-full 3xl:h-72 lg:flex lg:items-center justify-center md:p-2"
+                style={{ borderRadius: "20px" }}
+              >
+                <div className="relative mx-5 md:mx-3 mt-4 lg:mt-0">
+                  <div
+                    className="hover:bg-gray-200 flex justify-center overflow-hidden transition duration-500"
+                    onMouseEnter={handleMouseEnter}
+                    onMouseLeave={handleMouseLeave}
+                  >
+                    <Image
+                      src={DP}
```

Comment:
see if you can add an edit button/icon on the DP image so that mobile users have explicit indication that they can edit the photo